gobblin-runtime/src/main/java/org/apache/gobblin/runtime/JobContext.java

```diff
@@ -544,8 +544,7 @@ public static Optional<Class<? extends DataPublisher>> getJobDataPublisherClass(
    * or {@link ConfigurationKeys#PUBLISH_DATA_AT_JOB_LEVEL} is set to true.
    */
   public static boolean shouldCommitDataInJob(State state) {
-    boolean jobCommitPolicyIsFull =
-        JobCommitPolicy.getCommitPolicy(state.getProperties()) == JobCommitPolicy.COMMIT_ON_FULL_SUCCESS;
+    boolean jobCommitPolicyIsFull = JobCommitPolicy.getCommitPolicy(state) == JobCommitPolicy.COMMIT_ON_FULL_SUCCESS;
```

Comment:
huh never realized getProperties() actually creates a copy of the underlying properties object instead of a pointer, not that obvious by the method name compared to e.g. `getCopyOfProperties`

Reply:
true, the naming doesn't give any hints.  `State.getProperties` are actually synthesized from several underlying separate `Properties`, so to assemble them all together necessarily requires a copy.  the only way to avoid would be if `State` were immutable and itself based on an immutable data structure, which of course `java.util.Properties` is NOT.